posts/Swift-Optionals-Are-Monads-Too.md

```diff
@@ -0,0 +1,264 @@
+Monads, Functors, and Applicative Functors oh my! Have you ever wondered what any of these concepts are and why they are useful? Let's dive into some explanations and then we can see how useful they really can be in your daily programming.
+  
+# Swift's Optional Type
+
+So how does Swift define Optional? Opening up the generated Swift header we see the following:
+
+```swift
+/// A type that can represent either a `Wrapped` value or `nil`, the absence
+/// of a value.
+public enum Optional<Wrapped> : _Reflectable, NilLiteralConvertible {
+    case None
+    case Some(Wrapped)
+    /// Construct a `nil` instance.
+    public init()
+    /// Construct a non-`nil` instance that stores `some`.
+    public init(_ some: Wrapped)
+    /// If `self == nil`, returns `nil`.  Otherwise, returns `f(self!)`.
+    @warn_unused_result
+    public func map<U>(@noescape f: (Wrapped) throws -> U) rethrows -> U?
+    /// Returns `nil` if `self` is `nil`, `f(self!)` otherwise.
+    @warn_unused_result
+    public func flatMap<U>(@noescape f: (Wrapped) throws -> U?) rethrows -> U?
+    /// Create an instance initialized with `nil`.
+    public init(nilLiteral: ())
+}
+```
+
+Reading the first comment in the source code gives us an awesome definition of what an Optional really is. 
+
+> A type that can represent either a Wrapped value or nil, the absence of a value.
+
+Put in other words, an Optional is a container that wraps the concept of having a value or not.
+
+So why is this useful? Because Swift is a strongly typed language it makes us code for both cases of an Optional, having a value (.Some) or no value at all (.None). When we declare variables with let or var and make them Optional, we are telling the compiler that our variable may contain a value or not. And any subsequent access to our variable will require us to check that it contains a value before performing actions on the wrapped value.
```

Comment:
This may read better:

So why is this useful? Because Swift is a strongly typed language `that` makes us code for both cases of an Optional.  That is to say, having a value (.Some) or no value at all (.None). When we declare variables with let or var and make them Optional, we are telling the compiler that our variable may contain a value or not. `Any` subsequent access to our variable will require us to check that it contains a value before performing actions on the wrapped value.